packages/Parallel/Application/ParallelFileProcessor.php

```diff
@@ -129,14 +115,26 @@ public function process(
 
         $timeoutInSeconds = $this->parameterProvider->provideIntParameter(Option::PARALLEL_JOB_TIMEOUT_IN_SECONDS);
 
-        for ($i = 0; $i < $numberOfProcesses; ++$i) {
-            // nothing else to process, stop now
-            if ($jobs === []) {
-                break;
-            }
+        $workerCommandLineFactory = $this->workerCommandLineFactory;
+        $this->processSpawner = function(
```

Comment:
any need to have it as a property?

Reply:
you can make it a local variable, and pass it by reference in the `use` list to use it inside

Reply:
I actively wanted to avoid having it as a local variable, because there are to many of them which makes reading and changing the code a lot more complex. I would have preferred to make this an actual method in the class, but this would have needed more rewrites

Reply:
if a private method is too expensive, to me a local variable is a better fit here than a property, as it's a callable used only locally there